lighthouse-core/lib/i18n/README.md

```diff
@@ -0,0 +1,305 @@
+    (parenthesis). Note that `[link text] (https://...)` is NOT VALID because of
+    the space and will not be converted to a link.
+
+    Example usage:
+
+    ```javascript
+    const UIStrings = {
+      description: 'The value of ... [Learn More](https://google.com/)',
+    };
+    ```
+
+### ICU Replacement Syntax
+
+ICU syntax is used throughout Lighthouse strings, and they are specified
+directly in UIStrings. They follow 1 of 3 flavors.
+
+*   Direct replacement
+
+    This is simply a direct replacement of text into a string. Often used for
+    Proper Nouns, code, or other text that is dynamic and added at runtime, but
+    requires no additional formatting.
+
+    Example usage:
+
+    ```javascript
+    const UIStrings = {
+        didntCollectScreenshots: `Chrome didn't .... ({errorCode})`,
+    };
+    ```
+
+    ICU replacements must use a JSDoc type syntax to specify an example for
+    direct ICU replacements.
+
+    *   To specify the description, use
+
+        `@description <description>`
+
+    *   To specify an example for an ICU replacement, use
+
+        `@example {<ICU variable name>} <example for ICU replacement>`
+
+    ```javascript
+    const UIStrings = {
+        /**
+         * @description Error message explaining ...
+         * @example {errorCode} NO_SPEEDLINE_FRAMES
+         */
+        didntCollectScreenshots: `Chrome didn't .... ({errorCode})`,
+    };
+    ```
+
+*   Complex replacement
+
+    When more complex numerical ICU replacement is needed the syntax is mostly
+    the same as direct replacement. This is often used when replcaing times, or
+    percentages.
+
+    Note: these complex ICU formats are automatically given example values based
+    on their ICU format as specified in `collect-strings.js`.  Therefore a normal 
+    description string can be used.
+
+    ```javascript
+    const UIStrings = {
+        displayValueText: 'Interactive at {timeInMs, number, seconds} s',
+    };
+    ```
+
+TODO(exterkamp): explain all the comments and where they go/what they become.
+
+TODO(exterkamp): explain why we can't use some ICU like number formatting.
+
+### The pipeline
+
+The translation pipeline has 2 distinct stages, the collection and translation
+is done pre-compile time, and the replacement is done at runtime.
+
+#### Translation Pipeline (pre-compile)
+
+1.  `file_with_UIStrings.js` this is where strings start. Optimized for
+    programmer ease of use and not for machine parsing. Uses ICU syntax and
+    markdown control characters inline.
+
+2.  `yarn i18n:collect-strings` collects all UIStrings, and generates the
+    pre-locales. Does some parsing to make sure that common mistakes are
+    avoided.
+
+3.  `pre-locale/en-US.json` this is the well formatted _machine parsable_
+    fileset that is uploaded to be translated, i.e. they use $placeholder$
+    syntax instead of ICU. These will never be used by the internal i18n system,
+    they are solely used to send to translators.
+
```

Comment:
maybe add a step about sending to translators and getting back `pre-locale` files for all languages? Otherwise isn't clear in the next step where they came from